network/server.go

```diff
@@ -419,7 +451,14 @@ func (s *Server) delPeer(id peer.ID) {
 	s.peersLock.Lock()
 	defer s.peersLock.Unlock()
 
-	delete(s.peers, id)
+	peer, ok := s.peers[id]
```

Comment:
You can inline this init within the if statement

Reply:
If anybody reading this, this is the commit that fixes it:

https://github.com/0xPolygon/polygon-sdk/pull/326/commits/d5b6d98bc3210e3a5cc730859d015ceb6d63857a